src/site/content/en/blog/javascript-this/index.md

```diff
@@ -0,0 +1,356 @@
+});
+```
+
+{% endCompare %}
+
+I avoid using `this` in cases like above, and instead:
+
+{% Compare 'better' %}
+
+```js
+element.addEventListener('click', (event) => {
+  // Grab it from the parent scope:
+  console.log(element);
+  // Or from the event object:
+  console.log(event.currentTarget);
+});
+```
+
+{% endCompare %}
+
+## Otherwise, if the function is called via a parent object (`parent.func()`): {: #object-member }
+
+```js
+const obj = {
+  someMethod() {
+    return this;
+  },
+};
+
+// Logs `true`:
+console.log(obj.someMethod() === obj);
+```
+
+In this case the function is called as a member of `obj`, so `this` will be `obj`. This happens at call-time, so the link is broken if the function is called without its parent object, or with a different parent object:
+
+```js
+const {someMethod} = obj;
+// Logs `false`:
+console.log(someMethod() === obj);
+
+const anotherObj = {someMethod};
+// Logs `false`:
+console.log(anotherObj.someMethod() === obj);
+// Logs `true`:
+console.log(anotherObj.someMethod() === anotherObj);
+```
+
+`someMethod() === obj` is false because `someMethod` _isn't_ called as a member of `obj`. You might have encountered this gotcha when trying something like this:
+
+```js
+const $ = document.querySelector;
+// TypeError: Illegal invocation
+const el = $('.some-element');
+```
+
+This breaks because the implementation of `querySelector` looks it its own `this` value and expects it to be a DOM node of sorts, and the above breaks that connection. To achieve the above correctly:
+
+```js
+const $ = document.querySelector.bind(document);
+// Or:
+const $ = (...args) => document.querySelector(...args);
+```
+
+{% Aside 'warning' %}
+Don't transplant a function onto an object just to set `this` to some value unrelated to the parent object; it's usually unexpected and it's why `this` gets such a bad reputation. Consider passing the value as an argument instead; it's more explicit, and works with arrow functions.
+{% endAside %}
+
+## Otherwise, if the function or parent scope is in strict mode: {: #strict }
+
+```js
+function someFunction() {
+  'use strict';
+  return this;
+}
+
+// Logs `true`:
+console.log(someFunction() === undefined);
+```
+
+In this case, the value of `this` is undefined. `'use strict'` isn't needed in the function if the parent scope is in [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode).
+
+{% Aside 'warning' %}
+Don't rely on this. I mean, there are easier ways to get an `undefined` value 😀.
+{% endAside %}
+
+## Otherwise: {: #otherwise }
+
+```js
+function someFunction() {
+  return this;
+}
+
+// Logs `true`:
+console.log(someFunction() === globalThis);
+```
+
+In this case, the value of `this` is the global object.
+
+{% Aside 'warning' %}
+Avoid using `this` to reference the global object. Instead, use [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis), which is much more explicit.
```

Comment:
IMHO `globalThis` deserves a more prominent mention — there’s an opportunity to explain its name and how it relates to `this`, which (based on the flamewars that happened when the proposal renamed `global` to `globalThis`) lots of developers are confused about.